redbot/cogs/mod/settings.py

```diff
@@ -85,35 +79,155 @@ async def hierarchy(self, ctx: commands.Context):
+            msg = ""
+            msg += _("Warn mention spam: {num_mentions}\n").format(
+                num_mentions=_("{num} mentions").format(num=warn_mention_spam)
+                if warn_mention_spam
+                else _("No")
+            )
+            msg += _("Kick mention spam: {num_mentions}\n").format(
+                num_mentions=_("{num} mentions").format(num=kick_mention_spam)
+                if kick_mention_spam
+                else _("No")
+            )
+            msg += _("Ban mention spam: {num_mentions}\n").format(
+                num_mentions=_("{num} mentions").format(num=ban_mention_spam)
+                if ban_mention_spam
+                else _("No")
+            )
+
+            await ctx.send(box(msg))
+
+    @mentionspam.command()
+    @commands.guild_only()
+    async def warn(self, ctx: commands.Context, max_mentions: int = 0):
+        """
+        Sets the autowarn conditions for mention spam.
+
+        Users will be warned if they send any messages with contain more than
+        `<max_mentions>` mentions.
+
+        `<max_mentions>` must be at least 2. Set to 0 to disable.
+        """
+        mention_spam = await self.config.guild(ctx.guild).mention_spam.all()
+        if not max_mentions:
+            if not mention_spam["warn"]:
+                return await ctx.send(_("Autowarn for mention spam is already disabled."))
+            await self.config.guild(ctx.guild).mention_spam.warn.set(False)
+            return await ctx.send(_("Autowarn for mention spam disabled."))
+
+        if max_mentions < 2:
+            return await ctx.send(_("`<max_mentions>` must be 2 or higher to autowarn."))
+
+        mismatch_message = ""
+
+        if mention_spam["kick"]:
+            if max_mentions >= mention_spam["kick"]:
+                mismatch_message += _("\nAutowarn is equal or higher than autokick.")
+
+        if mention_spam["ban"]:
+            if max_mentions >= mention_spam["ban"]:
+                mismatch_message += _("\nAutowarn is equal to or higher than autoban.")
+
+        await self.config.guild(ctx.guild).mention_spam.warn.set(max_mentions)
+        await ctx.send(
+            _(
+                "Autowarn for mention spam enabled. "
+                "Anyone mentioning {max_mentions} or more different people "
+                "in a single message will be autowarned.\n{mismatch_message}"
+            ).format(max_mentions=max_mentions, mismatch_message=mismatch_message)
+        )
+
+    @mentionspam.command()
     @commands.guild_only()
-    async def banmentionspam(self, ctx: commands.Context, max_mentions: int = 0):
+    async def kick(self, ctx: commands.Context, max_mentions: int = 0):
+        """
+        Sets the autokick conditions for mention spam.
+
+        Users will be kicked if they send any messages with contain more than
+        `<max_mentions>` mentions.
+
+        `<max_mentions>` must be at least 3. Set to 0 to disable.
```

Comment:
Why does max mentions have to be at least three, besides to allow for autowarn to function at two? Couldn't autowarn be disabled and autokick be set to two?

Reply:
I forgot why I set it to 3, I just know the ban part (as you commented as well on) has an arbitrary number to lock it down from the previous coding. We could remove the number lock, or at least require all of them to be 2 (or even 1) or more.